website/static/js/share/utils.js

```diff
@@ -65,15 +71,33 @@ var search = function(vm) {
     vm.page = 0;
     vm.results = [];
 
-    History.pushState({query: vm.query()}, 'OSF | SHARE', '?q=' + vm.query());
+    History.pushState({
+        optionalFilters: vm.optionalFilters,
+        requiredFilters: vm.requiredFilters,
+        query: vm.query(),
+        sort: vm.sort()
+    }, 'OSF | SHARE', '?'+ $.param({
+        'q': vm.query(),
+        'required': vm.requiredFilters.join('|'),
```

Comment:
Maybe move this up into a separate variable and conditionally add optional and required to keep the url as clean as possible.